### ISSUE TITLE
Crash on changing routing method

### ISSUE THREAD

**Issue by Lineflyer** [cgeo/cgeo#13413]
### Describe your problem!

User reports on support mail, that c:geo crashes each time he changes the routing mode e.g. from "Car" to "Bike" or "Walk".

### How to reproduce?

Unclear

### Actual result after these steps?

```
08-20 16:07:36.546 14322 14414 E AndroidRuntime: FATAL EXCEPTION: RxComputationThreadPool-7
08-20 16:07:36.546 14322 14414 E AndroidRuntime: Process: cgeo.geocaching, PID: 14322
08-20 16:07:36.546 14322 14414 E AndroidRuntime: java.lang.NullPointerException: Attempt to invoke virtual method 'void cgeo.geocaching.models.Route.calculateNavigationRoute()' on a null object reference
08-20 16:07:36.546 14322 14414 E AndroidRuntime: 	at cgeo.geocaching.maps.mapsforge.v6.NewMap.lambda$routingModeChanged$6(NewMap.java:529)
08-20 16:07:36.546 14322 14414 E AndroidRuntime: 	at cgeo.geocaching.maps.mapsforge.v6.NewMap.lambda$routingModeChanged$6$NewMap(Unknown Source:0)
08-20 16:07:36.546 14322 14414 E AndroidRuntime: 	at cgeo.geocaching.maps.mapsforge.v6.-$$Lambda$NewMap$xAZ8GC96q-fXhBfWZfcSrtt-zxM.call(Unknown Source:6)
08-20 16:07:36.546 14322 14414 E AndroidRuntime: 	at cgeo.geocaching.maps.Tracks.traverse(Tracks.java:74)
08-20 16:07:36.546 14322 14414 E AndroidRuntime: 	at cgeo.geocaching.maps.mapsforge.v6.NewMap.lambda$routingModeChanged$7(NewMap.java:528)
08-20 16:07:36.546 14322 14414 E AndroidRuntime: 	at cgeo.geocaching.maps.mapsforge.v6.NewMap.lambda$routingModeChanged$7$NewMap(Unknown Source:0)
08-20 16:07:36.546 14322 14414 E AndroidRuntime: 	at cgeo.geocaching.maps.mapsforge.v6.-$$Lambda$NewMap$oVSkRikDbf5pZAHUh7JewQd_nxo.run(Unknown Source:2)
08-20 16:07:36.546 14322 14414 E AndroidRuntime: 	at cgeo.geocaching.utils.AndroidRxUtils.lambda$andThenOnUi$2(AndroidRxUtils.java:91)
08-20 16:07:36.546 14322 14414 E AndroidRuntime: 	at cgeo.geocaching.utils.-$$Lambda$AndroidRxUtils$bqzWo9nlvU86BvboB-9jhDWp5vc.run(Unknown Source:4)
08-20 16:07:36.546 14322 14414 E AndroidRuntime: 	at io.reactivex.rxjava3.internal.schedulers.ScheduledRunnable.run(ScheduledRunnable.java:65)
08-20 16:07:36.546 14322 14414 E AndroidRuntime: 	at io.reactivex.rxjava3.internal.schedulers.ScheduledRunnable.call(ScheduledRunnable.java:56)
08-20 16:07:36.546 14322 14414 E AndroidRuntime: 	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
08-20 16:07:36.546 14322 14414 E AndroidRuntime: 	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:301)
08-20 16:07:36.546 14322 14414 E AndroidRuntime: 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
08-20 16:07:36.546 14322 14414 E AndroidRuntime: 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
08-20 16:07:36.546 14322 14414 E AndroidRuntime: 	at java.lang.Thread.run(Thread.java:929)
08-21 
```

### Expected result after these steps?

no crash

### Reproducible

Unclear

### c:geo Version

2022.09.04

### System information

```text
## System information

c:geo version: 2022.09.04

Device:
-------
- Device type: ELE-L29 (ELE-L29EEA, HUAWEI)
- Available processors: 8
- Android version: 10
- Android build: ELE-L29 10.1.0.150(C431E22R2P5)
- Screen resolution: 1080x2139px (360x713dp)
- Pixel density: 3.0
- System font scale: 1.0 / used scale: 1.0
- Sailfish OS detected: false
- Google Play services: enabled - 22.26.15 (100408-461192076)

Sensor and location:
-------
- Low power mode: inactive
- Compass capabilities: yes
- Rotation vector sensor: present
- Orientation sensor: present
- Magnetometer & Accelerometer sensor: present
- Direction sensor used: orientation

Program settings:
-------
- Settings: v9, Count:225
- Set language: de
- System date format: dd.MM.yy
- Time zone: GMT+02:00
- Debug mode active: no
- Last backup: never
- Routing mode: Car
- Live map mode: true
- OSM multi-threading: true / threads: 4
- Map: OpenStreetMap.de
  - Id: cgeo.geocaching.maps.mapsforge.MapsforgeMapProvider$OsmdeMapSource
  - Atts: OpenStreetMap DE, map data OpenStreetMap contributors
  - Theme: Elevate.zip:Elements.xml

Filters:
-------
- Hide waypoints: original parking visited
- LIVE: Keine ([:inconclusive=false:advanced=true]status)
- OFFLINE: Keine ([:inconclusive=false:advanced=false]AND(type;difficulty_terrain:d=-:d=-:t=-:t=-;status))

Stored Filters (#0):

Services:
-------
- Geocaching sites enabled:
   geocaching.com: Logged in (Anmeldung OK) / PREMIUM
   Geocaching.com Adventure Lab
- Geocaching.com date format: dd.MM.yyyy
- Last login error on geocaching.com: communication error (2022-08-14 17-09)
- Last successful login on geocaching.com: 2022-09-06 14-21
- Routing: internal / BRouter installed: false
- Installed c:geo plugins:  none

Permissions & paths:
-------
- Fine location permission: granted
- Write external storage permission: granted
- System internal c:geo dir: /data/user/0/cgeo.geocaching (76,7 GB free) v3 internal isDir(14 entries)
- Legacy User storage c:geo dir: /storage/emulated/0/cgeo (76,7 GB free) v3 external non-removable isDir(6 entries)
- Geocache data: /storage/emulated/0/Android/data/cgeo.geocaching/files/GeocacheData (76,7 GB free) v3 external non-removable isDir(1080 entries)
- Internal theme sync (is turned off): /data/user/0/cgeo.geocaching/MapThemeData (76,7 GB free) v3 internal isDir(0 entries)
- Public Folders: #11
  - BASE: /cgeo (User-Defined)[/cgeo[DOCUMENT#0:p-content://com.android.externalstorage.documents/tree/primary%3Acgeo::]] (Uri: content://com.android.externalstorage.documents/tree/primary%3Acgeo/document/primary%3Acgeo, Av:true, files:>=57, dirs:>=12, totalFileSize:>=603,5 MB, free space: 76,7 GB, files on device: 13278715)
  - OFFLINE_MAPS: Speicherkarte/Karten (User-Defined)[Speicherkarte/Karten[DOCUMENT#0:p-content://com.android.externalstorage.documents/tree/0123-4567%3AKarten::]] (Uri: content://com.android.externalstorage.documents/tree/0123-4567%3AKarten/document/0123-4567%3AKarten, Av:true, files:6, dirs:1, totalFileSize:3,7 GB, free space: 112,0 GB, files on device: 0)
  - OFFLINE_MAP_THEMES: Speicherkarte/Karten (User-Defined)[Speicherkarte/Karten[DOCUMENT#0:p-content://com.android.externalstorage.documents/tree/0123-4567%3AKarten::]] (Uri: content://com.android.externalstorage.documents/tree/0123-4567%3AKarten/document/0123-4567%3AKarten, Av:true, files:6, dirs:1, totalFileSize:3,7 GB, free space: 112,0 GB, files on device: 0)
  - LOGFILES: /cgeo/logfiles (Default)[/cgeo/logfiles[PERSISTABLE_FOLDER(BASE)#1:p-content://com.android.externalstorage.documents/tree/primary%3Acgeo::/logfiles]] (Uri: content://com.android.externalstorage.documents/tree/primary%3Acgeo/document/primary%3Acgeo%2Flogfiles, Av:true, files:2, dirs:0, totalFileSize:65,2 KB, free space: 76,7 GB, files on device: 13278715)
  - GPX: /cgeo/gpx (Default)[/cgeo/gpx[PERSISTABLE_FOLDER(BASE)#1:p-content://com.android.externalstorage.documents/tree/primary%3Acgeo::/gpx]] (Uri: content://com.android.externalstorage.documents/tree/primary%3Acgeo/document/primary%3Acgeo%2Fgpx, Av:true, files:40, dirs:0, totalFileSize:21,4 MB, free space: 76,7 GB, files on device: 13278715)
  - BACKUP: /cgeo/backup (Default)[/cgeo/backup[PERSISTABLE_FOLDER(BASE)#1:p-content://com.android.externalstorage.documents/tree/primary%3Acgeo::/backup]] (Uri: content://com.android.externalstorage.documents/tree/primary%3Acgeo/document/primary%3Acgeo%2Fbackup, Av:true, files:>=19, dirs:>=6, totalFileSize:>=728,8 MB, free space: 76,7 GB, files on device: 13278715)
  - FIELD_NOTES: /cgeo/field-notes (Default)[/cgeo/field-notes[PERSISTABLE_FOLDER(BASE)#1:p-content://com.android.externalstorage.documents/tree/primary%3Acgeo::/field-notes]] (Uri: content://com.android.externalstorage.documents/tree/primary%3Acgeo/document/primary%3Acgeo%2Ffield-notes, Av:true, files:4, dirs:0, totalFileSize:4,8 KB, free space: 76,7 GB, files on device: 13278715)
  - SPOILER_IMAGES: /cgeo/GeocachePhotos (Default)[/cgeo/GeocachePhotos[PERSISTABLE_FOLDER(BASE)#1:p-content://com.android.externalstorage.documents/tree/primary%3Acgeo::/GeocachePhotos]] (Uri: content://com.android.externalstorage.documents/tree/primary%3Acgeo/document/primary%3Acgeo%2FGeocachePhotos, Av:true, files:0, dirs:3, totalFileSize:0 B, free space: 76,7 GB, files on device: 13278715)
  - ROUTING_BASE: /cgeo/routing (Default)[/cgeo/routing[PERSISTABLE_FOLDER(BASE)#1:p-content://com.android.externalstorage.documents/tree/primary%3Acgeo::/routing]] (Uri: content://com.android.externalstorage.documents/tree/primary%3Acgeo/document/primary%3Acgeo%2Frouting, Av:true, files:10, dirs:1, totalFileSize:251,1 MB, free space: 76,7 GB, files on device: 13278715)
  - ROUTING_TILES: /cgeo/routing/segments4 (Default)[/cgeo/routing/segments4[PERSISTABLE_FOLDER(ROUTING_BASE)#1:p-content://com.android.externalstorage.documents/tree/primary%3Acgeo::/routing/segments4]] (Uri: content://com.android.externalstorage.documents/tree/primary%3Acgeo/document/primary%3Acgeo%2Frouting%2Fsegments4, Av:true, files:3, dirs:0, totalFileSize:251,0 MB, free space: 76,7 GB, files on device: 13278715)
  - TEST_FOLDER: [Legacy]/data/user/0/cgeo.geocaching/files/unittest (Default)[/data/user/0/cgeo.geocaching/files/unittest[FILE#1:p-file:///data/user/0/cgeo.geocaching/files::/unittest]] (Uri: file:///data/user/0/cgeo.geocaching/files/unittest, Av:true, files:0, dirs:0, totalFileSize:0 B, free space: 76,7 GB, files on device: -1)
- Map render theme path: Elevate.zip:Elements.xml
- PersistedDocumentUris: #1
- UNUSED_TRACK: null
- Persisted Uri Permissions: #2
  - content://com.android.externalstorage.documents/tree/primary%3Acgeo (29. März 2021, 09:09):RW
  - content://com.android.externalstorage.documents/tree/0123-4567%3AKarten (21. Sept. 2021, 21:08):RW
- Database: /data/user/0/cgeo.geocaching/databases/data (v99, Size:42,6 MB) on system internal storage

--- End of system information ---
```

### Additional Information

Support ticket 744188

**Comment by moving-bits** [comment]
Hm, my hope was that this got solved with our 2022.09.04 release, but apparently that's not the case.
(We had a user with a similar issue, using 2022.08.21 release, and for him it seems to be resolved with 2022.09.04)

**Comment by moving-bits** [comment]
Have added a null check with PR #13435
